File: src/Domain/WAccount.Domain.Models/Transaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;
using WAccount.Domain.Models.Enumerators;

namespace WAccount.Domain.Models
{
    public class Transaction : BaseEntity
    {
        [Required]
        public int UserId { get; set; }
        public virtual UserAccount User { get; set; }

        [Required]
        public TransactionType Type { get; set; }

        [Required]
        public decimal Amount { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime Scheduling { get; set; }

        public TransactionResult Result { get; set; }
    }
}

File: front/src/app/transactions/transaction/transaction.ts
import { TransactionType } from './transaction.type';
import { TransactionResult } from './transaction.result';

export interface Transaction {
    userId: number;
    user: null;
    type: TransactionType;
    amount: number;
    scheduling: Date;
    result: TransactionResult;
    id: number;
    updatedAt: Date;  
}